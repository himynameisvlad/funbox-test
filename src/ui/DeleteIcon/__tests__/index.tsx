import { shallow } from 'enzyme';
import * as React from 'react';
import renderer from 'react-test-renderer';

import { DeleteIcon } from '../index';

describe('DeleteIcon - ui component', () => {
  it('renders correctly', () => {
    const onClick = () => null;
    const tree = renderer
      .create(<DeleteIcon onClick={onClick} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('calls onClick prop when button is clicked', () => {
    const onClickSpy = jest.fn();
    const wrapper = shallow(<DeleteIcon onClick={onClickSpy} />);
    const button = wrapper.find('button').first();
    button.simulate('click');

    expect(onClickSpy).toHaveBeenCalledTimes(1);
  });
})
