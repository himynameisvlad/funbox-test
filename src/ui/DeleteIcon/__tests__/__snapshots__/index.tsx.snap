// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteIcon - ui component renders correctly 1`] = `
<button
  className="fb-delete-icon"
  onClick={[Function]}
/>
`;
